### PR TITLE
Implement Profile Page

### PR DESCRIPTION
**Pull request by emichaud998**
Pull requests into this repository require the following checks to be completed by the submitter and reviewers.
Changes in this PR:
Added functionality to the capability receiver to pull out all the supported profiles from a conformance resource or capability statement, and created the profile tab on the shiny dashboard to display this information.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [Lantern 362 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-362)
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

### CODE REVIEW

File: shinydashboard/lantern/server.R [hunk 1]
```diff
@@ -168,6 +177,8 @@ function(input, output, session) { #nolint
 
   show_resource_checkbox <- reactive(input$side_menu %in% c("resource_tab"))
 
+  show_profiles_checkbox <- reactive(input$side_menu %in% c("profile_tab"))
```
Comment (bmath10) [inline shinydashboard/lantern/server.R:180]: If possible, I'd name this something like "show_profiles_filters"

File: shinydashboard/lantern/server.R [hunk 2]
```diff
@@ -416,4 +475,73 @@ function(input, output, session) { #nolint
               options = list("plugins" = list("remove_button"), "create" = TRUE, "persist" = FALSE))
     }
   })
+
+  #                                          #
+  #   Display Resource and Profile Filters   #
+  #                                          #
+
+  output$profile_filter_tab <- renderUI({
```
Comment (bmath10) [inline shinydashboard/lantern/server.R:483]: shouldn't this be the resource_filter_tab and the one below the profile_filter_tab?

File: capabilityreceiver/pkg/capabilityhandler/supportedProfiles.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+		}
+
+		supportedProfiles = append(supportedProfiles, profileInfo)
+
+	}
+
+	return supportedProfiles
+}
+
+// getCapabilityStatementProfiles stores all the profiles found in the Capability statement rest->resource->profile field
```
Comment (bmath10) [inline capabilityreceiver/pkg/capabilityhandler/supportedProfiles.go:52]: I think this should say `rest->resource->supportedProfile`

File: capabilityreceiver/pkg/capabilityhandler/capabilityreceiver_test.go [hunk 4]
```diff
@@ -573,6 +573,46 @@ func Test_RunSupportedResourcesChecks(t *testing.T) {
...
+	expectedResource := ""
+	th.Assert(t, supportedProfiles[0].ProfileURL == expectedURL, fmt.Sprintf("Expected ProfileURL to be %s, was %s", supportedProfiles[0].ProfileURL, expectedURL))
+	th.Assert(t, supportedProfiles[0].ProfileName == expectedName, fmt.Sprintf("Expected ProfileName to be %s, was %s", supportedProfiles[0].ProfileName, expectedName))
+	th.Assert(t, supportedProfiles[0].Resource == expectedResource, fmt.Sprintf("Expected resource to be an empty string, was %s", supportedProfiles[0].Resource))
+	expectedName = "U.S. Data Access Framework (DAF) Patient Profile"
+	expectedURL = "http://hl7.org/fhir/StructureDefinition/daf-patient"
+	expectedResource = ""
+	th.Assert(t, supportedProfiles[7].ProfileURL == expectedURL, fmt.Sprintf("Expected ProfileURL to be %s, was %s", supportedProfiles[7].ProfileURL, expectedURL))
+	th.Assert(t, supportedProfiles[7].ProfileName == expectedName, fmt.Sprintf("Expected ProfileName to be %s, was %s", supportedProfiles[7].ProfileName, expectedName))
+	th.Assert(t, supportedProfiles[0].Resource == expectedResource, fmt.Sprintf("Expected resource to be an empty string, was %s", supportedProfiles[7].Resource))
```
Comment (bmath10) [inline capabilityreceiver/pkg/capabilityhandler/capabilityreceiver_test.go:594]: should be checking `supportedProfiles[7].Resource`

File: endpointmanager/pkg/endpointmanager/fhirendpointinfo_test.go [hunk 5]
```diff
@@ -435,6 +505,29 @@ func Test_FHIREndpointInfoEqual(t *testing.T) {
...
+	if endpointInfo1.Equal(endpointInfo2) {
+		t.Errorf("Did not expect endpointInfo1 to equal endpointInfo 2. SupportedProfiles should be different. %+v vs %+v", endpointInfo1.SupportedProfiles, endpointInfo2.SupportedProfiles)
+	}
+	endpointInfo2.SupportedProfiles = endpointInfo1.SupportedProfiles
+
+	endpointInfo1.SupportedProfiles = make([]SupportedProfile, 0)
+	if endpointInfo1.Equal(endpointInfo2) {
+		t.Errorf("Did not expect endpointInfo1 to equal endpointInfo 2. SupportedProfiles should be different. %+v vs %+v", endpointInfo1.SupportedProfiles, endpointInfo2.SupportedProfiles)
+	}
+	endpointInfo1.SupportedProfiles = endpointInfo2.SupportedProfiles
```
Comment (bmath10) [inline endpointmanager/pkg/endpointmanager/fhirendpointinfo_test.go:529]: I'd also just write a check where they do equal each other, just to cover all of our bases!

File: shinydashboard/lantern/server.R (hunk 2 above)
Comment (emichaud998) [reply shinydashboard/lantern/server.R:483]: Yeah, no idea how I swapped these by accident! Good catch!

**Review by bmath10 — APPROVED** [review]

**Pull request merged**